### PR TITLE
Tensorflow 2 switch

### LINKED ISSUE #20: Correct imports in experiments notebooks
The experiments notebook have leftover imports.

### PR DESCRIPTION
**Pull request by zaccharieramzi**
This PR allows to switch from keras to tf2 (still using the keras API).

Notebooks have been reviewed and their imports are now corrected.

TF datasets have been implemented, but they are very slow. The only promising one, is the tfio based using directly hdf5 c++ API, but it is currently unusable because of https://github.com/tensorflow/io/issues/745. It also needs to be extended to zero-filled and special kiki.

Other notable changes:
- `fftshift` is now used directly from tf (thanks to https://github.com/tensorflow/tensorflow/pull/31409 !)
- adversarial training is still in an unstable state, especially with regards to the monitoring. We now use the gradient tape system to allow to get rid of the weird d on g structure (inspiration from https://github.com/RaphaelMeudec/tf2-deblurgan)

Important: not everything has been tested... (I know this is bad but I want to move forward with this).

This fixes #20 and #34 .

### CODE REVIEW

**Comment by zaccharieramzi** [comment]
Finally I was having bad thoughts and tested some stuff, at least the pdnet training script and the validation notebook. Everything working, with some issues to open:
- get rid of `fit_generator` to use `fit`.
- get the `h5py` warning fixed when opening HDF5 files (this because for now we have to stick to sequences).
